addon/utils/selection.js

```diff
@@ -63,24 +62,27 @@ export default {
    * @param {Function} itemComparator - comparator for items
    */
   /* eslint-disable complexity */
-  range (items, selectedItems, item, rangeState, itemComparator) {
+  range (items, selectedItems, item, rangeState, itemComparator, itemKey) {
     // If an anchor isn't set, then set the anchor and exit
     const rangeAnchor = rangeState['anchor']
-    if (isNone(rangeAnchor)) {
+
+    const anchor = items.findIndex(currentItem => itemComparator(currentItem, rangeState['anchor']))
+    // If anchor is -1 then it was a anchor from a previous page that we can not find, so reset
+    if (Ember.isEmpty(rangeAnchor) || anchor === -1) {
       // Range select is always a positive selection (no deselect)
       rangeState['anchor'] = item
 
       // New anchor, clear any previous endpoint
       rangeState['endpoint'] = Ember.Object.create()
 
-      // Add the anchor to the selected items
-      selectedItems.pushObject(item)
+      // Add the anchor to the selected items if not already in it from a previous page
```

Comment:
In pagination, the way this code is set up, your anchors should be reset.
If you are on page 2, do some shift clicking.
Switch to page 1, do a shift click on a already selected item. You will add this item into selectedItems due to there was no check.